packages/flutter/lib/src/widgets/scrollable_helpers.dart

```diff
@@ -460,17 +460,17 @@ class ScrollAction extends ContextAction<ScrollIntent> {
         return true;
       }());
 
-      if (primaryScrollController.position.context.notificationContext == null
-          && Scrollable.maybeOf(primaryScrollController.position.context.notificationContext!) == null) {
+      if (primaryScrollController.position.context.notificationContext != null) {
+        state = Scrollable.maybeOf(primaryScrollController.position.context.notificationContext!);
+      }
```

Comment:
I think it'd be good to set a local variable here, to further reduce the amount of null-asserting.

```suggestion
      final BuildContext? notificationContext = primaryScrollController.position.context.notificationContext;
      if (notificationContext != null) {
        state = Scrollable.maybeOf(notificationContext);
      }
```

Reply:
Done.